engine/src/main/java/org/pentaho/di/job/Job.java

```diff
@@ -375,6 +379,11 @@ public void setRepository( Repository rep ) {
       copyParametersFrom( jobMeta );
       activateParameters();
 
+      // Add the UUID to log4j MDC so it can be inserted in log lines
+      if( uuid != null ) {
```

Comment:
First Glance Question: in which case would a uuid be null? is that ever expected, and if not should there be error handling?

Reply:
Good question. We do sanitize the input in such a way that UUID is either passed by us or generated in the **Params** class constructor. But since we don't know the full scope of the **Job** class usage, I'd say it makes sense to make this validation.

Reply:
Yeah, that was my logic. In the normal worker nodes flow we don't need to worry about it, but because this is a pretty "core" class that's used in lots of different ways, I wanted to keep the changes small/safe.

Reply:
Are you sure we want to make the change in job.java and not in the CommandExecutor class. My only concern is that we are adding import to log4j in a class which is using the own logging classes

Reply:
Are we introducing dependency to log4j?

Reply:
The log4j jars are already included for the project, we add the import of course, but no new jars needed to be added to the pom for this change.

We can't add the UUIDs in just the executor because the MDC is thread specific and the jobs run in their own thread -- this changes is adding the value so that the UUID is visible in the context where the job logs are written.